src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3094,12 +3094,21 @@ private void AddSynthesizedRecordMembersIfNecessary(MembersAndInitializersBuilde
 
             return;
 
-            SynthesizedRecordConstructor addCtor(RecordDeclarationSyntax declWithParameters)
+            SynthesizedRecordConstructor? addCtor(RecordDeclarationSyntax declWithParameters)
             {
                 Debug.Assert(declWithParameters.ParameterList is object);
                 var ctor = new SynthesizedRecordConstructor(this, declWithParameters, diagnostics);
-                members.Add(ctor);
-                return ctor;
+
+                if (ctor.ParameterCount == 1 && ctor.Parameters[0].Type.Equals(this))
+                {
+                    diagnostics.Add(ErrorCode.ERR_RecordAmbigCtor, ctor.Locations[0]);
+                    return null;
```

Comment:
>return null; [](start = 20, length = 12)

This doesn't feel right. I think we should suppress creation of synthesized copy constructor instead. The primary constructor is explicitly declared, we cannot just ignore the declaration even if it is erroneous.

Reply:
If we suppress the primary constructor, then we'll also get a warning for unused parameter. I think it doesn't really matter which one we suppress. In my mind the copy constructor is always there, and in this case prevents the primary constructor from being declared. I don't see a problem.

---
In reply to: [543389753](https://github.com/dotnet/roslyn/pull/49975#discussion_r543389753) [](ancestors = 543389753)

Reply:
> If we suppress the primary constructor, then we'll also get a warning for unused parameter

I am not following. It looks like you are suppressing it.

---
In reply to: [543463726](https://github.com/dotnet/roslyn/pull/49975#discussion_r543463726) [](ancestors = 543463726,543389753)

Reply:
Whatever unexpected warning we get, should be suppressed, of course. But that should not come at the "expense of symbols"

> In my mind the copy constructor is always there, and in this case prevents the primary constructor from being declared. I don't see a problem.

I do. The primary constructor is clearly declared. So, there should be a corresponding symbol for it. I am also not suggesting to suppress copy constructor. All we need to do is to detect an ambiguity and report it.

---
In reply to: [543465253](https://github.com/dotnet/roslyn/pull/49975#discussion_r543465253) [](ancestors = 543465253,543463726,543389753)

Reply:
> I am also not suggesting to suppress copy constructor.

To clarify. I did suggest this as an alternative at the beginning because that constructor is synthesized and I would be comfortable with not synthesizing something "illegal" as opposed to not creating a symbol for something explicitly declared in code. However, I would be fine with leaving that symbol along as well. We could get in a situation of both symbols present when copy constructor is declared explicitly, therefore, the pipeline should be able to operate reliably with both symbols present.

---
In reply to: [543470457](https://github.com/dotnet/roslyn/pull/49975#discussion_r543470457) [](ancestors = 543470457,543465253,543463726,543389753)